dsl/Prop.cc

```diff
@@ -300,4 +300,31 @@ vector<unique_ptr<ast::Expression>> Prop::replaceDSL(core::MutableContext ctx, a
     return ret;
 }
 
+void Prop::patchDSL(core::MutableContext ctx, ast::ClassDef *klass) {
+    UnorderedMap<ast::Expression *, vector<unique_ptr<ast::Expression>>> replaceNodes;
+    for (auto &stat : klass->rhs) {
+        auto *send = ast::cast_tree<ast::Send>(stat.get());
+        if (send == nullptr) {
+            continue;
+        }
+        auto nodes = replaceDSL(ctx, send);
```

Comment:
Oh nice, I really like that you managed to keep the `replaceDSL` API and re-use it.
That makes this code much easier to reason about. Nice job.

Reply:
do you mean you prefer that I don't return an std::optional?

Reply:
resolved: no.